### ISSUE TITLE
gunicorn+nginx SSL issues with unix socket.

### ISSUE THREAD

**Issue by 133794m3r** [benoitc/gunicorn#2398]
I know that in #2062 that the person's issue was resolved by setting proxy_redirect to being off. I am, using unix sockets(as I had done with flask+uwsgi). I am using gunicorn b/c uwsgi can't find the callable(for whatever reason) but gunicorn works fine locally and thus I figured I'd use it also on the server.

Below is the scrollback from it not wanting to work. It's the same error as the person who was sending SSL traffic to gunicorn but since they were passing data via http instead of a unix socket I don't think this issue applies to me also as I'm not proxying traffic like that.

```
gunicorn capstone_project.wsgi --log-level debug --timeout 120 --workers=3 --threads=3 --worker-connections=1000 -b unix:/tmp/test.sock -u 33 -g 33
[2020-08-09 03:27:16 +0000] [7392] [DEBUG] Current configuration:
  config: None
  bind: ['unix:/tmp/test.sock']
  backlog: 2048
  workers: 3
  worker_class: sync
  threads: 3
  worker_connections: 1000
  max_requests: 0
  max_requests_jitter: 0
  timeout: 120
  graceful_timeout: 30
  keepalive: 2
  limit_request_line: 4094
  limit_request_fields: 100
  limit_request_field_size: 8190
  reload: False
  reload_engine: auto
  reload_extra_files: []
  spew: False
  check_config: False
  preload_app: False
  sendfile: None
  reuse_port: False
  chdir: /home/macarthur/ctf_test
  daemon: False
  raw_env: []
  pidfile: None
  worker_tmp_dir: None
  user: 33
  group: 33
  umask: 0
  initgroups: False
  tmp_upload_dir: None
  secure_scheme_headers: {'X-FORWARDED-PROTOCOL': 'ssl', 'X-FORWARDED-PROTO': 'https', 'X-FORWARDED-SSL': 'on'}
  forwarded_allow_ips: ['127.0.0.1']
  accesslog: None
  disable_redirect_access_to_syslog: False
  access_log_format: %(h)s %(l)s %(u)s %(t)s "%(r)s" %(s)s %(b)s "%(f)s" "%(a)s"
  errorlog: -
  loglevel: debug
  capture_output: False
  logger_class: gunicorn.glogging.Logger
  logconfig: None
  logconfig_dict: {}
  syslog_addr: udp://localhost:514
  syslog: False
  syslog_prefix: None
  syslog_facility: user
  enable_stdio_inheritance: False
  statsd_host: None
  dogstatsd_tags: 
  statsd_prefix: 
  proc_name: None
  default_proc_name: capstone_project.wsgi
  pythonpath: None
  paste: None
  on_starting: <function OnStarting.on_starting at 0x7f89b71367b8>
  on_reload: <function OnReload.on_reload at 0x7f89b71368c8>
  when_ready: <function WhenReady.when_ready at 0x7f89b71369d8>
  pre_fork: <function Prefork.pre_fork at 0x7f89b7136ae8>
  post_fork: <function Postfork.post_fork at 0x7f89b7136bf8>
  post_worker_init: <function PostWorkerInit.post_worker_init at 0x7f89b7136d08>
  worker_int: <function WorkerInt.worker_int at 0x7f89b7136e18>
  worker_abort: <function WorkerAbort.worker_abort at 0x7f89b7136f28>
  pre_exec: <function PreExec.pre_exec at 0x7f89b71510d0>
  pre_request: <function PreRequest.pre_request at 0x7f89b71511e0>
  post_request: <function PostRequest.post_request at 0x7f89b7151268>
  child_exit: <function ChildExit.child_exit at 0x7f89b7151378>
  worker_exit: <function WorkerExit.worker_exit at 0x7f89b7151488>
  nworkers_changed: <function NumWorkersChanged.nworkers_changed at 0x7f89b7151598>
  on_exit: <function OnExit.on_exit at 0x7f89b71516a8>
  proxy_protocol: False
  proxy_allow_ips: ['127.0.0.1']
  keyfile: None
  certfile: None
  ssl_version: 2
  cert_reqs: 0
  ca_certs: None
  suppress_ragged_eofs: True
  do_handshake_on_connect: False
  ciphers: None
  raw_paste_global_conf: []
  strip_header_spaces: False
[2020-08-09 03:27:16 +0000] [7392] [INFO] Starting gunicorn 20.0.4
[2020-08-09 03:27:16 +0000] [7392] [DEBUG] Arbiter booted
[2020-08-09 03:27:16 +0000] [7392] [INFO] Listening at: unix:/tmp/test.sock (7392)
[2020-08-09 03:27:16 +0000] [7392] [INFO] Using worker: threads
[2020-08-09 03:27:16 +0000] [7395] [INFO] Booting worker with pid: 7395
[2020-08-09 03:27:16 +0000] [7396] [INFO] Booting worker with pid: 7396
[2020-08-09 03:27:16 +0000] [7397] [INFO] Booting worker with pid: 7397
[2020-08-09 03:27:16 +0000] [7392] [DEBUG] 3 workers
[2020-08-09 03:28:18 +0000] [7395] [DEBUG] Ignored premature client disconnection. No more data after: b'\x00\x87\x03\x00\x0c\x00QUERY_STRING\x00\x00\x0e\x00REQUEST_METHOD\x03\x00GET\x0c\x00CONTENT_TYPE\x00\x00\x0e\x00CONTENT_LENGTH\x00\x00\x0b\x00REQUEST_URI\x01\x00/\t\x00PATH_INFO\x01\x00/\r\x00DOCUMENT_ROOT \x00/home/macarthur/capstone_project\x0f\x00SERVER_PROTOCOL\x08\x00HTTP/1.1\x0e\x00REQUEST_SCHEME\x05\x00https\x05\x00HTTPS\x02\x00on\x0b\x00REMOTE_ADDR\x0e\x00216.145.95.123\x0b\x00REMOTE_PORT\x05\x0035608\x0b\x00SERVER_PORT\x03\x00443\x0b\x00SERVER_NAME\x11\x00crimsonstrife.com\t\x00HTTP_HOST\x11\x00crimsonstrife.com\x0f\x00HTTP_CONNECTION\n\x00keep-alive\x12\x00HTTP_CACHE_CONTROL\t\x00max-age=0\x08\x00HTTP_DNT\x01\x001\x1e\x00HTTP_UPGRADE_INSECURE_REQUESTS\x01\x001\x0f\x00HTTP_USER_AGENTh\x00Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.92 Safari/537.36\x0b\x00HTTP_ACCEPT|\x00text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9\x13\x00HTTP_SEC_FETCH_SITE\x04\x00none\x13\x00HTTP_SEC_FETCH_MODE\x08\x00navigate\x13\x00HTTP_SEC_FETCH_USER\x02\x00?1\x13\x00HTTP_SEC_FETCH_DEST\x08\x00document\x14\x00HTTP_ACCEPT_ENCODING\x11\x00gzip, deflate, br\x14\x00HTTP_ACCEPT_LANGUAGE\x0e\x00en-US,en;q=0.9'
[2020-08-09 03:29:18 +0000] [7395] [DEBUG] Ignored premature client disconnection. No more data after: b"\x00\x1a\x03\x00\x0c\x00QUERY_STRING\x00\x00\x0e\x00REQUEST_METHOD\x03\x00GET\x0c\x00CONTENT_TYPE\x00\x00\x0e\x00CONTENT_LENGTH\x00\x00\x0b\x00REQUEST_URI\x0c\x00/favicon.ico\t\x00PATH_INFO\x0c\x00/favicon.ico\r\x00DOCUMENT_ROOT \x00/home/macarthur/capstone_project\x0f\x00SERVER_PROTOCOL\x08\x00HTTP/1.1\x0e\x00REQUEST_SCHEME\x05\x00https\x05\x00HTTPS\x02\x00on\x0b\x00REMOTE_ADDR\x0e\x00216.145.95.123\x0b\x00REMOTE_PORT\x05\x0035608\x0b\x00SERVER_PORT\x03\x00443\x0b\x00SERVER_NAME\x11\x00crimsonstrife.com\t\x00HTTP_HOST\x11\x00crimsonstrife.com\x0f\x00HTTP_CONNECTION\n\x00keep-alive\x0f\x00HTTP_USER_AGENTh\x00Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.92 Safari/537.36\x08\x00HTTP_DNT\x01\x001\x0b\x00HTTP_ACCEPT'\x00image/webp,image/apng,image/*,*/*;q=0.8\x13\x00HTTP_SEC_FETCH_SITE\x0b\x00same-origin\x13\x00HTTP_SEC_FETCH_MODE\x07\x00no-cors\x13\x00HTTP_SEC_FETCH_DEST\x05\x00empty\x0c\x00HTTP_REFERER\x1a\x00https://crimsonstrife.com/\x14\x00HTTP_ACCEPT_ENCODING\x11\x00gzip, deflate, br\x14\x00HTTP_ACCEPT_LANGUAGE\x0e\x00en-US,en;q=0.9"
```

nginx config file for site's enabled.
```
server {
	listen 80 default_server;
	listen [::]:80 default_server;
	root /home/macarthur/capstone_project;
	location /static{
		alias /home/macarthur/capstone_project/static;
	}
	location /{
		include uwsgi_params;
		#uwsgi_pass localhost:8000;
		uwsgi_pass unix:/tmp/test.sock;
	}
}
server {
	root /home/macarthur/capstone_project;
	location /static{
		alias /home/macarthur/capstone_project/static;
	}
	location /{
		include uwsgi_params;
		#uwsgi_pass localhost:8000;
		uwsgi_pass unix:/tmp/test.sock;
	}

    server_name crimsonstrife.com; # managed by Certbot

    listen [::]:443 ssl ipv6only=on; # managed by Certbot
    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/crimsonstrife.com/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/crimsonstrife.com/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

}
server {
    if ($host = crimsonstrife.com) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


	listen 80 ;
	listen [::]:80 ;
    server_name crimsonstrife.com;
    return 404; # managed by Certbot


}
```

**Comment by benoitc** [comment]
why closing it? did you solve your issue?

**Comment by 133794m3r** [comment]
I decided to just go back to using uwsgi as most people use the http socket with gunicorn. And also the fact that I probably would be setting it up as a reverse proxy meant that the other answers were likely correct.

I was already using uwsgi for my 2 flask applications on the server and I decided it'd be best to wait till I had the time to really test it locally to fix it. Plus I figured I'd be the only one who was ever dumb enough to not realize that nginx is still acting like a proxy even with the socket connection.
